doc/user/content/transform-data/dataflow-troubleshooting.md

```diff
@@ -479,3 +479,26 @@ WHERE
     AND mda.address[1] = dataflows.dataflow_address
     AND mdo.id = dataflows.dataflow_operator;
 ```
+
+## I dropped an index, why haven't my plans and dataflows changed?
+If you drop an index from the catalog, but it is depended on by downstream objects, then it will continue to be maintained and take up resources until all those dependent objects are dropped or altered, or Materialize is restarted.
+
+<!-- Todo: add reference to the console once it's available there. -->
+To see if you have any residual dataflows on a given cluster, you can run the following query
+```sql
+SET CLUSTER TO <clusterName>;
+SELECT s.name AS residualIndexName, s.id as dataflowId
+FROM mz_internal.mz_dataflow_arrangement_sizes AS s
+INNER JOIN mz_internal.mz_compute_exports AS ce ON ce.dataflow_id = s.id
+LEFT JOIN mz_catalog.mz_objects AS o ON o.id = ce.export_id
+WHERE o.id IS NULL;
+```
+
+<!-- Todo: Is there a way to list the dependencies? -->
```

Comment:
I think this would have to be a recursive query on `mz_compute_dependencies`. Maybe we should consider adding `mz_compute_transitive_dependencies` to match `mz_transitive_dependencies`?

Reply:
Ah you're right! I assumed based on the docs for `mz_compute_dependencies` that all rows in that table needed an `object_id` that could join to one of the `mz_catalog` tables like `mz_objects` or `mz_indexes`. But it looks like when we remove the row from `mz_indexes`/`mz_objects`, we don't cascade and drop the row from  
`mz_compute_dependencies` and `mz_compute_exports`.

For future reference, my test setup:
* Created webhook source `testing123`
* `create index testing_i on testing123 (body);` - id `u94`
* `create view testing_v as select body from testing123;`
* `create index testing_i_2 on testing_v (body);` - id `u95 `
* `DROP INDEX testing_i`
This produces an orphaned dataflow, confirming via the query I put in the docs.

`select * from mz_internal.mz_compute_dependencies where dependency_id like 'u%';
returns rows with `u94` in them (the orphaned dataflow), but there's nothing to join the id to, since it's been cleaned up from `mz_objects` and `mz_indexes` already.
That can be joined against `select * from mz_internal.mz_compute_exports` on `export_id`, which same as the first query can be joined to `mz_internal.mz_dataflow_arrangement_sizes` on `dataflow_id`.

Reply:
I put up a commit for a non-recursive query that just returns immediate dependencies.
Here's the recursive one
```
WITH MUTUALLY RECURSIVE
  reach(object_id text, referenced_object_id text) AS (
    SELECT object_id, dependency_id FROM mz_internal.mz_compute_dependencies
    UNION
    SELECT x, z FROM reach r1(x, y) JOIN reach r2(y, z) USING(y)
  )
SELECT do.id AS dependent_object_id, do.name AS dependent_object_name, db.name AS dependent_object_database, s.name AS dependent_object_schema
FROM reach
LEFT JOIN mz_catalog.mz_objects AS do ON reach.object_id = do.id
LEFT JOIN mz_catalog.mz_schemas s ON do.schema_id = s.id 
LEFT JOIN mz_catalog.mz_databases AS db ON s.database_id = db.id
WHERE referenced_object_id = <former_object_id from above>;
```
Do you think it's better to use this one instead?

Reply:
After thinking about this more, it seems direct dependencies are probably most interesting to users, so the non-recursive version probably fine.